sympy/solvers/solveset.py

```diff
@@ -1117,3 +1120,514 @@ def linsolve(system, *symbols):
     # Return solutions
     solution = FiniteSet(tuple(solution))
     return solution
+
+
+##############################################################################
+# ------------------------------nonlinsolve ---------------------------------#
+##############################################################################
+
+
+def substitution(system, symbols, result, known_symbols, exclude, all_symbols):
```

Comment:
>Does this needs to be available in public API?

I didn't understand Public API ?

It is helper function for `nonlinsolve` but able to solve simple system.